spec/unit/guard_interpreter/resource_guard_interpreter_spec.rb

```diff
@@ -29,28 +29,105 @@
+      parent_resource.guard_interpreter(:script)
+      expect { guard_interpreter }.not_to raise_error
     end
-    
+
     it "allows the guard interpreter to be set to Chef::Resource::PowershellScript derived indirectly from Chef::Resource::Script" do
-      resource.guard_interpreter(:powershell_script)
-      expect { Chef::GuardInterpreter::ResourceGuardInterpreter.new(resource, "echo hi", nil) }.not_to raise_error
+      parent_resource.guard_interpreter(:powershell_script)
+      expect { guard_interpreter }.not_to raise_error
     end
-    
+
     it "raises an exception if guard_interpreter is set to a resource not derived from Chef::Resource::Script" do
-      resource.guard_interpreter(:file)
-      expect { Chef::GuardInterpreter::ResourceGuardInterpreter.new(resource, "echo hi", nil) }.to raise_error(ArgumentError)
+      parent_resource.guard_interpreter(:file)
+      expect { guard_interpreter }.to raise_error(ArgumentError, 'Specified guard interpreter class Chef::Resource::File must be a kind of Chef::Resource::Execute resource')
+    end
+
+    context "when the resource cannot be found for the platform" do
+      before do
+        expect(Chef::Resource).to receive(:resource_for_node).with(:foobar, node).and_return(nil)
+      end
+
+      it "raises an exception" do
+        parent_resource.guard_interpreter(:foobar)
+        expect { guard_interpreter }.to raise_error(ArgumentError, 'Specified guard_interpreter resource foobar unknown for this platform')
+      end
+    end
+
+    it "fails when parent_resource is nil" do
+      expect { Chef::GuardInterpreter::ResourceGuardInterpreter.new(nil, "exit 0", nil) }.to raise_error(ArgumentError, /Node for guard resource parent must not be nil/)
+    end
+
+  end
+
+  describe "#evaluate" do
+    let(:guard_interpreter) { Chef::GuardInterpreter::ResourceGuardInterpreter.new(parent_resource, "exit 0", {}) }
+    let(:parent_resource) do
+      parent_resource = Chef::Resource.new("execute resource", run_context)
+      parent_resource.guard_interpreter(:execute)
+      parent_resource
+    end
+
+    it "successfully evaluates the resource" do
+      expect(guard_interpreter.evaluate).to eq(true)
+    end
+
+    describe "Script command opts switch" do
+      let(:command_opts) { {} }
+      let(:guard_interpreter) { Chef::GuardInterpreter::ResourceGuardInterpreter.new(parent_resource, "exit 0", command_opts) }
+
+      context "resource is a Script" do
+        context "and guard_interpreter is :script" do
+          let(:parent_resource) do
+            parent_resource = Chef::Resource::Script.new("resource", run_context)
+            parent_resource.guard_interpreter(:script)
+            parent_resource
+          end
+
+          it "merges to :code" do
+            expect(command_opts).to receive(:merge).with({:code => "exit 0"}).and_call_original
+            guard_interpreter.evaluate
```

Comment:
I don't think you should be calling valuate without mocking the parts that shell out